apps/condo/domains/notification/schema/_internalSendNotificationNewMobileAppVersionService.js

```diff
@@ -0,0 +1,151 @@
+                            },
+                        },
+                        chunkSize: 50,
+                        chunkProcessor: (/** @type {RemoteClient[]} */ chunk) => {
+                            const rcWithLowerBuild = []
+
+                            if (buildVersion) {
+                                for (const rc of chunk) {
+                                    if (get(rc, 'meta.Build', buildVersion) < buildVersion) rcWithLowerBuild.push(`rc:${rc.id}`)
+                                }
+                            }
+
+                            return rcWithLowerBuild
+                        },
+                    })
+                } else {    //The case when a push is sent to suitable devices of a specific or several organizations
+                    const userIds = await loadListByChunks({
+                        context: adminContext,
+                        list: Resident,
+                        where: {
+                            organization: {
+                                id_in: organizationIds,
+                            },
+                        },
+                        chunkSize: 50,
+                        chunkProcessor: (/** @type {Resident[]} */ chunk) => {
+                            const userIds = []
+                            chunk.map(resident => {
+                                const userId = get(resident, 'user.id')
+                                if (userId) userIds.push(userId)
+                            })
+                            return userIds
+                        },
+                    })
+
+                    remoteClientKeys = await loadListByChunks({
+                        context: adminContext,
+                        list: RemoteClient,
+                        where: {
+                            devicePlatform: platform,
+                            deletedAt: null,
+                            owner: {
+                                type: userType,
+                                id_in: userIds,
+                            },
+                        },
+                        chunkSize: 50,
+                        chunkProcessor: (/** @type {RemoteClient[]} */ chunk) => {
+                            const rcWithLowerBuild = []
+
+                            if (buildVersion) {
+                                for (const rc of chunk) {
+                                    if (get(rc, 'meta.Build', buildVersion) < buildVersion) rcWithLowerBuild.push(`rc:${rc.id}`)
+                                }
+                            }
+
+                            return rcWithLowerBuild
+                        },
+                    })
+                }
+
+                const messageBatch = await MessageBatch.create(adminContext, {
+                    dv,
+                    sender,
+                    messageType: MOBILE_APP_UPDATE_AVAILABLE_MESSAGE_PUSH_TYPE,
+                    targets: remoteClientKeys,
```

Comment:
Are you ok for batch with giga-json? Since target can be 100k+ ids...

Reply:
Yes. Butch has no restrictions as far as I know.